activestorage/app/models/active_storage/blob/representable.rb

```diff
@@ -25,9 +25,9 @@ module ActiveStorage::Blob::Representable
   #
   # Raises ActiveStorage::InvariableError if ImageMagick cannot transform the blob. To determine whether a blob is
   # variable, call ActiveStorage::Blob#variable?.
-  def variant(transformations)
+  def variant(transformations, delivery_method = nil)
```

Comment:
Not liking this. It's a smell that the delivery method has to get passed around like this. Whatever the final return is, there should be a way to call a URL method on it that specifies the delivery method. Not when creating the variant or the preview.